src/lib.rs

```diff
@@ -915,6 +1016,32 @@ impl Deref for PayeePubKey {
 	}
 }
 
+impl ExpiryTime {
+	pub fn from_seconds(seconds: u64) -> Result<ExpiryTime, CreationError> {
+		if seconds <= SYSTEM_TIME_MAX_UNIX_TIMESTAMP - MAX_EXPIRY_TIME {
```

Comment:
Did you mean to just make sure its <= MAX_EXPIRY_TIME?

Reply:
Yes, at first I used `i64::MAX` as maximum for `SystemTime` and just defined that both `PositiveTimestamp` and `ExpiryTime` could be up to `i64::MAX / 2` big. But then I looked at other platforms and had to decrease the maximum to `i32::MAX` which made dividing by 2 no longer feasible and I had to define two thresholds. 

I probably didn't think much about it and just replaced every `SYSTEM_TIME_MAX_UNIX_TIMESTAMP / 2` with `SYSTEM_TIME_MAX_UNIX_TIMESTAMP - MAX_EXPIRY_TIME` not taking the asymmetry into account.